src/java/org/apache/cassandra/db/commitlog/CommitLog.java

```diff
@@ -282,6 +285,11 @@ public void recover(String path) throws IOException
         recoverPath(path, false);
     }
 
+    public void setCommitLogSegmentHandler(CommitLogSegmentHandler handler)
```

Comment:
Usually we inject custom behavior a bit differently.
We establish an interface, C* provides the default implementation and a factory that is capable of building instances of a custom implementation.
CNDB provides a custom implementation and its name via system property. See WriteCli or CoordinatorCli.

Reply:
I guess the main thing about it is the ability to turn off the new behavior by overriding the system property in the helm chart.